infer/environment.cc

```diff
@@ -1129,7 +1129,10 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
                 // by desugar and redacting the `SendResponse` so LSP features work
                 // more like developers expect.
                 const bool isDesugarTripleEqSend = send.fun == core::Names::tripleEq() && send.funLoc.empty();
-                if (lspQueryMatch && !isDesugarTripleEqSend) {
+                // Something like `X = Foo` will be rewritten to `X = Magic.<suggest-constant-type>(Foo)`
+                // We don't want to report a send response for the <suggest-constant-type> call here.
+                const bool isSuggestConstantType = send.fun == core::Names::suggestConstantType();
+                if (lspQueryMatch && !isDesugarTripleEqSend && !isSuggestConstantType) {
```

Comment:
Minor nit: we might want to just have one variable, like `const bool ignoreSendForLSPQuery`. Instead of having to write

```cpp
                if (lspQueryMatch && !isDesugarTripleEqSend && !isSuggestConstantType) {
```

we could just write

```cpp
                if (lspQueryMatch && ignoreSendForLSPQuery) {
```